src/main/java/com/gmail/goosius/siegewar/utils/TownPeacefulnessUtil.java

```diff
@@ -265,19 +265,38 @@ public static void evaluatePeacefulTownOccupationAssignments() {
 		}
 	}
 
-	private static Nation getPrevailingNation(Set<Town> guardianTowns) {
-		//Find the largest guardian town & thus prevailing nation
-		Town topGuardianTown = null;
-		for(Town guardianTown: guardianTowns) {
-			if(topGuardianTown == null || guardianTown.getTownBlocks().size() > topGuardianTown.getTownBlocks().size()) {
-				topGuardianTown = guardianTown;
+	/*
+	 * Amplify the influence of home & enemy nations
+	 */
+	private static Map<Nation, Long> amplifyInfluenceOfHomeAndEnemyNations(Map<Nation, Long> nationInfluenceMap, Town peacefulTown) {
+		Map<Nation, Long> result = new HashMap<>(nationInfluenceMap);
+		long amplifiedValue;
+		Nation nationOfPeacefulTown;
+		if (peacefulTown.hasNation()) {
+			nationOfPeacefulTown = TownyAPI.getInstance().getTownNationOrNull(peacefulTown);
+			for (Map.Entry<Nation, Long> mapEntry : nationInfluenceMap.entrySet()) {
+				if (mapEntry.getKey() == nationOfPeacefulTown
+						|| mapEntry.getKey().hasEnemy(nationOfPeacefulTown)) {
+					amplifiedValue = mapEntry.getValue() * 1000000;
+					result.put(mapEntry.getKey(), amplifiedValue);
+				}
 			}
 		}
-		if(TownOccupationController.isTownOccupied(topGuardianTown)) {
-			return TownOccupationController.getTownOccupier(topGuardianTown);
-		} else {
-			return TownyAPI.getInstance().getTownNationOrNull(topGuardianTown);
+		return result;
+	}
+
+	private static Nation calculateNationWithStrongestInfluence(Map<Nation, Long> guardianNations) {
+		Map.Entry<Nation, Long> winningEntry = null;
```

Comment:
This might able to be simplied to just:
```java
		Map.Entry<Nation, Long> winningEntry = guardianNations.entrySet().stream()
			.max(Comparator.comparingLong(nation -> nation.getValue()))
			.get();
```

Reply:
- Sure I'll do that
- I need to start learning about this kind of pattern anyway.

Reply:
Update pushed, now requires re-approval